src/Interpreters/Streaming/CalculateDataStreamSemantic.h

```diff
@@ -13,10 +13,19 @@ namespace Streaming
 /// @brief Return whether current semantic can tracking changes:
 /// 1) Storage (VersionedKV/ChangelogKV/Changelog)
 /// 2) Or flat transformation / filtering data stream over 1) Storages
-bool canTrackChangesFromStorage(DataStreamSemanticEx input_data_stream_semantic);
+/// 3) Or changelog data stream
+bool canTrackChangesFromInput(DataStreamSemanticEx input_data_stream_semantic);
 
+/// Used for <stream> join <stream>
 bool isJoinResultChangelog(DataStreamSemanticEx left_data_stream_semantic, DataStreamSemanticEx right_data_stream_semantic);
 
+/// Used for <stream/table> join <stream/table>
```

Comment:
It is wierd to add this function. Only 'bool isJoinResultChangelog(DataStreamSemanticEx left_data_stream_semantic, DataStreamSemanticEx right_data_stream_semantic)' is enough. It is better to put the logic of this function into the origin 'isJoinResultChangelog' function.